### PR TITLE
Feature/cleanup warnings

### PR DESCRIPTION
**Pull request by robertjensen**
A number of updates, almost all related to running cinfdata in a modern linux-setup with a new version of python and php.

A large number of php-warnings has been handled, and a number of python3 fixes has also been implemented.

A few places, hard-coded references to the orginal cinfdata installation has been abstracted to a more general case.

### CODE REVIEW

File: common_functions_v2.php [hunk 1]
```diff
@@ -61,10 +63,10 @@ function xscale($max,$min,$manual,$log){
     @return array
   */
 function date_xscale($from="",$to="",$defaulthours=24){
-  $xscale["to"] = date('Y-m-d H:i',time() + 60); // Default, 1 minute into the future, to be shure get the whole plot                         
+  $xscale["to"] = date('Y-m-d H:i',time() + 60 * 60 * 24 * 30); // Default, 1 month into the feature
```
Comment (KennethNielsen) [inline common_functions_v2.php:66]: I think I agree with changing the deafult, but remember to get @Ejler in on it.

File: sym-files2/databasebackend.py [hunk 2]
```diff
@@ -166,38 +165,50 @@ def _process_data(self):
...
         if self.ggs['default_xscale'] == 'dat':
             return
-        # Run plugin(s) on data
-        self._plugins()
+
         # Convert the data to be as a function of ...
         if self.o['as_function_of']:
             self._as_function_of()
+
+        # Run plugin(s) on data
```
Comment (KennethNielsen) [inline sym-files2/databasebackend.py:177]: Are you sure this is what we want. To convert to as function of, before running plugins. I don't remember the design and use cases well enough to be sure

File: sym-files2/databasebackend.py [hunk 3]
```diff
@@ -209,7 +220,17 @@ def _plugins(self):
         query = 'SELECT input FROM plot_com_in WHERE id={0}'.\
             format(self.o['input_id'])
         self.cursor.execute(query)
-        input_settings = json.loads(self.cursor.fetchall()[0][0].decode())
+        result = self.cursor.fetchall()[0][0].decode()
+
+        # I do not fully understand when result is empty, generally it is
+        # wanted to move away from plot_com_in and plot_com_out and rather
+        # use redis. Until then, the following line ensures that the code
```
Comment (KennethNielsen) [inline sym-files2/databasebackend.py:227]: Agreed on moving to a better system for that

File: sym-files2/graphsettings.php [hunk 4]
```diff
@@ -72,12 +72,26 @@ function plot_settings($type, $params="", $ignore_invalid_type=False, $gs_file='
   if (!$type_found and !$ignore_invalid_type){
     return NULL;
   }
-  
-  
+
+  // Handle the replacement of {label} markers in the settings
+  // This is a rather primitive non-recursive (works only to 2 levels deep, and
+  // this only by explicit handling of second level) way of doing it. Some happy
+  // day we should re-write this
```
Comment (KennethNielsen) [inline sym-files2/graphsettings.php:79]: Agreed

**Review by KennethNielsen — APPROVED** [review]
Looks good to me. But you should be aware the my php-foo is getting rusty and was never that strong to begin with. So for the php parts I have mainly just glanced the code to make sure it looks sensible at the surface, other than that I rely on your testing that it is good.

There are a few comments, but they are mainly non critical clarifying questions.

File: sym-files2/databasebackend.py (hunk 2 above)
Comment (robertjensen) [reply sym-files2/databasebackend.py:177]: I am as sure as I think I am going to be. We thought about the various cases, an as far as I can see, this is generally the wanted behavior.

**Pull request merged**